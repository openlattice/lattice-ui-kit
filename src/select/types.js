// @flow

type ReactSelectValue = Object | Object[] | null;

type ReactSelectAction =
  | "select-option"
  | "deselect-option"
  | "remove-value"
  | "pop-value"
  | "set-value"
  | "clear"
  | "create-option";

type ReactSelectEvent = {
  action :ReactSelectAction;
  name ? :string;
};

export type {
  ReactSelectValue,
  ReactSelectAction,
  ReactSelectEvent,
};
