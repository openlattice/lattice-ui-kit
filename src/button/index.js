/*
 * @flow
 */

export { default as Button } from './src/components/Button';
export { default as IconButton } from './src/components/IconButton';
