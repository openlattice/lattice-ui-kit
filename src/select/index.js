export {
  StyledSelect as Select,
  StyledCreatable as Creatable
} from './src/components/Select';