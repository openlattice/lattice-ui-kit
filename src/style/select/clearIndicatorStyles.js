const clearIndicatorStyles = (base) => ({
  ...base,
  margin: '5px',
  padding: '0',
});

export default clearIndicatorStyles;
