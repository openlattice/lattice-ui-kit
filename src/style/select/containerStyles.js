const containerStyles = (base, state) => ({
  ...base,
  cursor: state.isDisabled ? 'not-allowed' : 'default',
  pointerEvents: 'auto', // not sure what this was for
  width: '100%',
});

export default containerStyles;
