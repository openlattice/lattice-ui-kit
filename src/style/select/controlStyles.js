import { NEUTRALS, PURPLES, WHITE } from '../../colors/src/Colors';

const controlStyles = (base, state) => {

  const { isFocused, isDisabled, selectProps } = state;

  let backgroundColor = isFocused ? WHITE : NEUTRALS[8];
  let border = isFocused ? `solid 1px ${PURPLES[1]}` : `solid 1px ${NEUTRALS[4]}`;

  if (selectProps && selectProps.borderless) {
    backgroundColor = 'transparent';
    border = 'none';
  }

  const hoverBgColor = (isDisabled || isFocused) ? WHITE : NEUTRALS[6];

  const style = {
    backgroundColor,
    border,
    borderRadius: '3px',
    boxShadow: 'none',
    fontSize: '14px',
    lineHeight: 1.5,
    minHeight: '40px',
    pointerEvents: isDisabled ? 'none' : 'auto', // not sure what this was for
    transition: 'background-color 0.2s ease-out, border-color 0.2s ease-out',
    ':hover': {
      border,
      backgroundColor: hoverBgColor,
    },
  };

  return { ...base, ...style };

};

export default controlStyles;
