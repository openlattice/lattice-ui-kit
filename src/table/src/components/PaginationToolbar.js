// @flow
import React from 'react';
import { faChevronLeft, faChevronRight } from '@fortawesome/pro-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { PaginationWrapper, RowPerPageWrapper } from './styled';
import { getRowsPerPageOptions } from './TableUtils';
import { Select } from '../../../select';
import { IconButton } from '../../../button';
import Label from '../../../label';

type Props = {
  count :number;
  page :number;
  rowsPerPage :number;
  rowsPerPageOptions ? :number[];
  setPage :(number :number) => void;
  setRowsPerPage :(number :number) => void;
};

const PaginationToolbar = (props :Props) => {
  const {
    count,
    page,
    rowsPerPage,
    rowsPerPageOptions,
    setPage,
    setRowsPerPage,
  } = props;

  const options = getRowsPerPageOptions(rowsPerPageOptions, count);
  const lastPage = Math.floor(count / rowsPerPage) - Number(!(count % rowsPerPage));

  const maxRowNumber = Math.min(rowsPerPage * (page + 1), count);
  const minRowNumber = Math.min(rowsPerPage * page + 1, count);
  const rowRange = `${minRowNumber} - ${maxRowNumber} of ${count}`;

  return (
    <PaginationWrapper>
      <Label subtle>Rows per page</Label>
      {
        (options.length > 1) && (
          <RowPerPageWrapper>
            <Select
                borderless
                defaultValue={options[0]}
                onChange={(rows) => {
                  setPage(0);
                  setRowsPerPage(rows);
                }}
                options={options}
                value={rowsPerPage}
                useRawValues />
          </RowPerPageWrapper>
        )
      }
      <Label id="row-range" subtle>{rowRange}</Label>
      <IconButton
          mode="subtle"
          icon={<FontAwesomeIcon icon={faChevronLeft} fixedWidth />}
          disabled={page <= 0}
          onClick={() => setPage(page - 1)} />
      <IconButton
          mode="subtle"
          icon={<FontAwesomeIcon icon={faChevronRight} fixedWidth />}
          disabled={page >= lastPage}
          onClick={() => setPage(page + 1)} />
    </PaginationWrapper>
  );
};

PaginationToolbar.defaultProps = {
  count: 0,
  page: 0,
  rowsPerPage: 5,
  rowsPerPageOptions: []
};

// $FlowFixMe
export default React.memo(PaginationToolbar);
