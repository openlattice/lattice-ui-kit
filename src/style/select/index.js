import clearIndicatorStyles from './clearIndicatorStyles';
import containerStyles from './containerStyles';
import controlStyles from './controlStyles';
import dropdownIndicatorStyles from './dropdownIndicatorStyles';
import indicatorSeparatorStyles from './indicatorSeparatorStyles';
import indicatorsContainerStyles from './indicatorsContainerStyles';
import menuListStyles from './menuListStyles';
import menuPortalStyles from './menuPortalStyles';
import menuStyles from './menuStyles';
import optionStyles from './optionStyles';
import selectTheme from './selectTheme';
import singleValueStyles from './singleValueStyles';
import valueContainerStyles from './valueContainerStyles';

const selectStyles = {
  clearIndicator: clearIndicatorStyles,
  container: containerStyles,
  control: controlStyles,
  dropdownIndicator: dropdownIndicatorStyles,
  indicatorSeparator: indicatorSeparatorStyles,
  indicatorsContainer: indicatorsContainerStyles,
  menu: menuStyles,
  menuList: menuListStyles,
  menuPortal: menuPortalStyles,
  option: optionStyles,
  singleValue: singleValueStyles,
  valueContainer: valueContainerStyles,
};

export {
  clearIndicatorStyles,
  containerStyles,
  controlStyles,
  dropdownIndicatorStyles,
  indicatorSeparatorStyles,
  indicatorsContainerStyles,
  menuListStyles,
  menuPortalStyles,
  menuStyles,
  optionStyles,
  selectStyles,
  selectTheme,
  singleValueStyles,
  valueContainerStyles,
};
