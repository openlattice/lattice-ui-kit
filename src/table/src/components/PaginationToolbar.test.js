import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import PaginationToolbar from './PaginationToolbar';
import { IconButton } from '../../../button';
import { Select } from '../../../select';

describe('PaginationToolbar', () => {

  describe('snapshots', () => {
    test('should match snaphot', () => {
      const wrapper = mount(
        <PaginationToolbar
            page={0}
            count={0} />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('props', () => {
    describe('setPage and setRowsPerPage', () => {
      let mockSetPage;
      let mockSetRowsPerPage;
      let wrapper;
      let buttons;
      beforeEach(() => {
        mockSetPage = jest.fn();
        mockSetRowsPerPage = jest.fn();
        wrapper = mount(
          <PaginationToolbar
              page={0}
              setPage={mockSetPage}
              setRowsPerPage={mockSetRowsPerPage} />
        );
        buttons = wrapper.find(IconButton);
      });
      test('should invoke setPage with page - 1 when back button is clicked', () => {
        const prevButton = buttons.get(0);

        expect(mockSetPage).toHaveBeenCalledTimes(0);
        prevButton.props.onClick();
        expect(mockSetPage).toHaveBeenCalledTimes(1);
        expect(mockSetPage.mock.calls[0][0]).toEqual(-1);
      });

      test('should invoke setPage with page + 1 when back button is clicked', () => {
        const nextButton = buttons.get(1);

        expect(mockSetPage).toHaveBeenCalledTimes(0);
        nextButton.props.onClick();
        expect(mockSetPage).toHaveBeenCalledTimes(1);
        expect(mockSetPage.mock.calls[0][0]).toEqual(1);
      });

      test('should setPage to 0 for Select onChange', () => {
        const select = wrapper.find(Select).get(0);

        expect(mockSetPage).toHaveBeenCalledTimes(0);
        select.props.onChange();
        expect(mockSetPage).toHaveBeenCalledTimes(1);
        expect(mockSetPage.mock.calls[0][0]).toEqual(0);
      });

      test('should setRowsPerPage when Select onChange', () => {
        const select = wrapper.find(Select).get(0);

        expect(mockSetRowsPerPage).toHaveBeenCalledTimes(0);
        select.props.onChange(20);
        expect(mockSetRowsPerPage).toHaveBeenCalledTimes(1);
        expect(mockSetRowsPerPage.mock.calls[0][0]).toEqual(20);
      });
    });
  });
});
