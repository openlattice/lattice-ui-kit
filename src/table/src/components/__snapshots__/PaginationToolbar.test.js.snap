// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationToolbar snapshots should match snaphot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
}

.c1 {
  color: #1f1f22;
  display: inline-block;
  font-size: 14px;
  font-stretch: normal;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 5px 5px 5px 0;
  color: #717175;
  font-size: 0.75rem;
  font-weight: 600;
  text-transform: uppercase;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
}

.c2::before {
  content: "";
  display: block;
  padding-bottom: 100%;
}

.c2 >:first-child {
  visibility: visible;
}

<PaginationToolbar
  count={0}
  page={0}
>
  <styled__PaginationWrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__PaginationWrapper-r3xwjs-1",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "align-items:center;display:flex;justify-content:flex-end;width:100%;",
            ],
          },
          "displayName": "styled__PaginationWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__PaginationWrapper-r3xwjs-1",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <Label
          id="row-range"
          subtle={true}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Label-sc-130fyca-0",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "color:",
                    "#1f1f22",
                    ";display:inline-block;font-size:14px;font-stretch:normal;font-style:normal;font-weight:",
                    [Function],
                    ";letter-spacing:normal;margin:5px 5px 5px 0;visibility:",
                    [Function],
                    ";",
                    [Function],
                    ";",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "Label",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Label-sc-130fyca-0",
                "target": "label",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            id="row-range"
            subtle={true}
          >
            <label
              className="c1"
              id="row-range"
            >
              0 - 0 of 0
            </label>
          </StyledComponent>
        </Label>
        <ForwardRef
          disabled={true}
          onClick={[Function]}
        >
          <IconButton
            disabled={true}
            forwardRef={null}
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(IconButton))
              className=""
              color="default"
              disabled={true}
              onClick={[Function]}
            >
              <ForwardRef(IconButton)
                className=""
                classes={
                  Object {
                    "colorInherit": "MuiIconButton-colorInherit",
                    "colorPrimary": "MuiIconButton-colorPrimary",
                    "colorSecondary": "MuiIconButton-colorSecondary",
                    "disabled": "Mui-disabled",
                    "edgeEnd": "MuiIconButton-edgeEnd",
                    "edgeStart": "MuiIconButton-edgeStart",
                    "label": "MuiIconButton-label",
                    "root": "MuiIconButton-root",
                    "sizeSmall": "MuiIconButton-sizeSmall",
                  }
                }
                color="default"
                disabled={true}
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ButtonBase))
                  centerRipple={true}
                  className="MuiIconButton-root Mui-disabled"
                  disabled={true}
                  focusRipple={true}
                  onClick={[Function]}
                >
                  <ForwardRef(ButtonBase)
                    centerRipple={true}
                    className="MuiIconButton-root Mui-disabled"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    disabled={true}
                    focusRipple={true}
                    onClick={[Function]}
                  >
                    <button
                      className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <IconButton__IconSquareWrapper
                          isLoading={false}
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "IconButton__IconSquareWrapper-g9q5gl-0",
                                  "isStatic": false,
                                  "lastClassName": "c2",
                                  "rules": Array [
                                    "align-items:center;display:flex;justify-content:center;width:auto;&::before{content:\\"\\";display:block;padding-bottom:100%;}>:first-child{visibility:",
                                    [Function],
                                    ";}",
                                  ],
                                },
                                "displayName": "IconButton__IconSquareWrapper",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "IconButton__IconSquareWrapper-g9q5gl-0",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            isLoading={false}
                          >
                            <div
                              className="c2"
                            >
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={true}
                                flip={null}
                                icon={
                                  Object {
                                    "icon": Array [
                                      256,
                                      512,
                                      Array [],
                                      "f053",
                                      "M231.293 473.899l19.799-19.799c4.686-4.686 4.686-12.284 0-16.971L70.393 256 251.092 74.87c4.686-4.686 4.686-12.284 0-16.971L231.293 38.1c-4.686-4.686-12.284-4.686-16.971 0L4.908 247.515c-4.686 4.686-4.686 12.284 0 16.971L214.322 473.9c4.687 4.686 12.285 4.686 16.971-.001z",
                                    ],
                                    "iconName": "chevron-left",
                                    "prefix": "far",
                                  }
                                }
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                swapOpacity={false}
                                symbol={false}
                                title=""
                                transform={null}
                              >
                                <svg
                                  aria-hidden="true"
                                  className="svg-inline--fa fa-chevron-left fa-w-8 fa-fw "
                                  data-icon="chevron-left"
                                  data-prefix="far"
                                  focusable="false"
                                  role="img"
                                  style={Object {}}
                                  viewBox="0 0 256 512"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M231.293 473.899l19.799-19.799c4.686-4.686 4.686-12.284 0-16.971L70.393 256 251.092 74.87c4.686-4.686 4.686-12.284 0-16.971L231.293 38.1c-4.686-4.686-12.284-4.686-16.971 0L4.908 247.515c-4.686 4.686-4.686 12.284 0 16.971L214.322 473.9c4.687 4.686 12.285 4.686 16.971-.001z"
                                    fill="currentColor"
                                    style={Object {}}
                                  />
                                </svg>
                              </FontAwesomeIcon>
                            </div>
                          </StyledComponent>
                        </IconButton__IconSquareWrapper>
                      </span>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(IconButton)>
            </WithStyles(ForwardRef(IconButton))>
          </IconButton>
        </ForwardRef>
        <ForwardRef
          disabled={true}
          onClick={[Function]}
        >
          <IconButton
            disabled={true}
            forwardRef={null}
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(IconButton))
              className=""
              color="default"
              disabled={true}
              onClick={[Function]}
            >
              <ForwardRef(IconButton)
                className=""
                classes={
                  Object {
                    "colorInherit": "MuiIconButton-colorInherit",
                    "colorPrimary": "MuiIconButton-colorPrimary",
                    "colorSecondary": "MuiIconButton-colorSecondary",
                    "disabled": "Mui-disabled",
                    "edgeEnd": "MuiIconButton-edgeEnd",
                    "edgeStart": "MuiIconButton-edgeStart",
                    "label": "MuiIconButton-label",
                    "root": "MuiIconButton-root",
                    "sizeSmall": "MuiIconButton-sizeSmall",
                  }
                }
                color="default"
                disabled={true}
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ButtonBase))
                  centerRipple={true}
                  className="MuiIconButton-root Mui-disabled"
                  disabled={true}
                  focusRipple={true}
                  onClick={[Function]}
                >
                  <ForwardRef(ButtonBase)
                    centerRipple={true}
                    className="MuiIconButton-root Mui-disabled"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    disabled={true}
                    focusRipple={true}
                    onClick={[Function]}
                  >
                    <button
                      className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <IconButton__IconSquareWrapper
                          isLoading={false}
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "IconButton__IconSquareWrapper-g9q5gl-0",
                                  "isStatic": false,
                                  "lastClassName": "c2",
                                  "rules": Array [
                                    "align-items:center;display:flex;justify-content:center;width:auto;&::before{content:\\"\\";display:block;padding-bottom:100%;}>:first-child{visibility:",
                                    [Function],
                                    ";}",
                                  ],
                                },
                                "displayName": "IconButton__IconSquareWrapper",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "IconButton__IconSquareWrapper-g9q5gl-0",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            isLoading={false}
                          >
                            <div
                              className="c2"
                            >
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={true}
                                flip={null}
                                icon={
                                  Object {
                                    "icon": Array [
                                      256,
                                      512,
                                      Array [],
                                      "f054",
                                      "M24.707 38.101L4.908 57.899c-4.686 4.686-4.686 12.284 0 16.971L185.607 256 4.908 437.13c-4.686 4.686-4.686 12.284 0 16.971L24.707 473.9c4.686 4.686 12.284 4.686 16.971 0l209.414-209.414c4.686-4.686 4.686-12.284 0-16.971L41.678 38.101c-4.687-4.687-12.285-4.687-16.971 0z",
                                    ],
                                    "iconName": "chevron-right",
                                    "prefix": "far",
                                  }
                                }
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                swapOpacity={false}
                                symbol={false}
                                title=""
                                transform={null}
                              >
                                <svg
                                  aria-hidden="true"
                                  className="svg-inline--fa fa-chevron-right fa-w-8 fa-fw "
                                  data-icon="chevron-right"
                                  data-prefix="far"
                                  focusable="false"
                                  role="img"
                                  style={Object {}}
                                  viewBox="0 0 256 512"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M24.707 38.101L4.908 57.899c-4.686 4.686-4.686 12.284 0 16.971L185.607 256 4.908 437.13c-4.686 4.686-4.686 12.284 0 16.971L24.707 473.9c4.686 4.686 12.284 4.686 16.971 0l209.414-209.414c4.686-4.686 4.686-12.284 0-16.971L41.678 38.101c-4.687-4.687-12.285-4.687-16.971 0z"
                                    fill="currentColor"
                                    style={Object {}}
                                  />
                                </svg>
                              </FontAwesomeIcon>
                            </div>
                          </StyledComponent>
                        </IconButton__IconSquareWrapper>
                      </span>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(IconButton)>
            </WithStyles(ForwardRef(IconButton))>
          </IconButton>
        </ForwardRef>
      </div>
    </StyledComponent>
  </styled__PaginationWrapper>
</PaginationToolbar>
`;
